Fails to build with `--locked` due to an old `time` crate version

grcov fails to build with the recent versions of Rust nightly (tested with `rustc 1.81.0-nightly (cf2df68d1 2024-07-01)`) due to time-rs/time#681:

```
error[E0282]: type annotations needed for `Box<_>`
  --> /home/m4tx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.23/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

   Compiling thiserror-impl v1.0.43
   Compiling clap_derive v4.3.2
   Compiling chrono-tz v0.6.1
   Compiling globwalk v0.8.1
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `grcov v0.8.19`, intermediate artifacts can be found at `/tmp/cargo-installw35YGR`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

Since the issue has been solved in `time` v0.3.36, it should be as easy as upgrading this one package in Cargo.lock (or running `cargo update`).

--- a/src/cobertura.rs
+++ b/src/cobertura.rs
@@ -4,9 +4,12 @@ use quick_xml::{
     Writer,
 };
 use rustc_hash::FxHashMap;
-use std::io::{BufWriter, Cursor, Write};
-use std::path::Path;
 use std::time::{SystemTime, UNIX_EPOCH};
+use std::{
+    fmt::Display,
+    io::{BufWriter, Cursor, Write},
+};
+use std::{fmt::Formatter, path::Path};
 use symbolic_common::Name;
 use symbolic_demangle::{Demangle, DemangleOptions};
 
@@ -220,10 +223,10 @@ enum ConditionType {
     Jump,
 }
 
-impl ToString for ConditionType {
-    fn to_string(&self) -> String {
-        match *self {
-            Self::Jump => String::from("jump"),
+impl Display for ConditionType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Jump => write!(f, "jump"),
         }
     }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -57,7 +57,7 @@ pub fn merge_results(result: &mut CovResult, result2: CovResult) -> bool {
             btree_map::Entry::Occupied(c) => {
                 let v = c.get().checked_add(execution_count).unwrap_or_else(|| {
                     warn_overflow = true;
-                    std::u64::MAX
+                    u64::MAX
                 });
 
                 *c.into_mut() = v;
@@ -140,7 +140,7 @@ fn rename_single_files(results: &mut [(String, CovResult)], stem: &str) {
     if let Some(parent) = Path::new(stem).parent() {
         for (file, _) in results.iter_mut() {
             if has_no_parent(file) {
-                *file = parent.join(&file).to_str().unwrap().to_string();
+                *file = parent.join(file.as_str()).to_str().unwrap().to_string();
             }
         }
     }

--- a/src/llvm_tools.rs
+++ b/src/llvm_tools.rs
@@ -245,33 +245,43 @@ mod tests {
         assert!(output_lcov
             .lines()
             .any(|line| line.contains("SF") && line.contains("src") && line.contains("main.rs")));
-        assert!(output_lcov.lines().any(|line| line.contains("FN:3")
-            && line.contains("rust_code_coverage_sample")
-            && line.contains("Ciao")));
+        if rustc_version::version_meta().unwrap().channel != rustc_version::Channel::Nightly {
+            assert!(output_lcov.lines().any(|line| line.contains("FN:3")
+                && line.contains("rust_code_coverage_sample")
+                && line.contains("Ciao")));
+        }
         assert!(output_lcov.lines().any(|line| line.contains("FN:8")
             && line.contains("rust_code_coverage_sample")
             && line.contains("main")));
-        assert!(output_lcov.lines().any(|line| line.contains("FNDA:0")
-            && line.contains("rust_code_coverage_sample")
-            && line.contains("Ciao")));
+        if rustc_version::version_meta().unwrap().channel != rustc_version::Channel::Nightly {
+            assert!(output_lcov.lines().any(|line| line.contains("FNDA:0")
+                && line.contains("rust_code_coverage_sample")
+                && line.contains("Ciao")));
+        } else {
+            assert!(output_lcov.lines().any(|line| line.contains("FNDA:1")
+                && line.contains("rust_code_coverage_sample")
+                && line.contains("main")));
+        }
         assert!(output_lcov.lines().any(|line| line.contains("FNDA:1")
             && line.contains("rust_code_coverage_sample")
             && line.contains("main")));
-        assert!(output_lcov.lines().any(|line| line == "FNF:2"));
+        if rustc_version::version_meta().unwrap().channel != rustc_version::Channel::Nightly {
+            assert!(output_lcov.lines().any(|line| line == "FNF:2"));
+        }
         assert!(output_lcov.lines().any(|line| line == "FNH:1"));
-        assert!(output_lcov.lines().any(|line| line == "DA:3,0"));
+        if rustc_version::version_meta().unwrap().channel != rustc_version::Channel::Nightly {
+            assert!(output_lcov.lines().any(|line| line == "DA:3,0"));
+        }
         assert!(output_lcov.lines().any(|line| line == "DA:8,1"));
         assert!(output_lcov.lines().any(|line| line == "DA:9,1"));
-        if rustc_version::version_meta().unwrap().channel != rustc_version::Channel::Nightly {
-            assert!(output_lcov.lines().any(|line| line == "DA:10,1"));
-        }
+        assert!(output_lcov.lines().any(|line| line == "DA:10,1"));
         assert!(output_lcov.lines().any(|line| line == "DA:11,1"));
         assert!(output_lcov.lines().any(|line| line == "DA:12,1"));
         assert!(output_lcov.lines().any(|line| line == "BRF:0"));
         assert!(output_lcov.lines().any(|line| line == "BRH:0"));
         if rustc_version::version_meta().unwrap().channel == rustc_version::Channel::Nightly {
             assert!(output_lcov.lines().any(|line| line == "LF:5"));
-            assert!(output_lcov.lines().any(|line| line == "LH:4"));
+            assert!(output_lcov.lines().any(|line| line == "LH:5"));
         } else {
             assert!(output_lcov.lines().any(|line| line == "LF:6"));
             assert!(output_lcov.lines().any(|line| line == "LH:5"));

--- a/src/reader.rs
+++ b/src/reader.rs
@@ -944,7 +944,7 @@ impl Gcno {
 
 impl GcovFunction {
     fn get_cycle_count(edges: &mut [GcovEdge], path: &[usize]) -> u64 {
-        let mut count = std::u64::MAX;
+        let mut count = u64::MAX;
         for e in path.iter() {
             count = cmp::min(count, edges[*e].cycles);
         }

--- a/tests/test.rs
+++ b/tests/test.rs
@@ -88,14 +88,7 @@ fn read_expected(
         "linux"
     };
 
-    let base_name = format!(
-        "expected{}",
-        if let Some(additional) = additional {
-            additional
-        } else {
-            ""
-        }
-    );
+    let base_name = format!("expected{}", additional.unwrap_or_default());
 
     let name_with_ver_and_os = format!(
         "{}_{}_{}_{}.{}",

